### PR TITLE
feat: add prometheus and observability docs

### PR DESCRIPTION
**Pull request by Qinyouzeng**

### CODE REVIEW

File: docs/framework/building-blocks/observability.md [hunk 1]
```diff
@@ -1,6 +1,188 @@
...
 title: 可观测性
-date: 2022/07/01
+date: 2022/12/07
 ---
 
-文档正在编写中...
\ No newline at end of file
+## 概念
+
+我们基于[Opentelemetry](https://opentelemetry.io/docs/)的可观测性接入，使用标准的OTLPExporter，[Logs](https://opentelemetry.io/docs/concepts/observability-primer/#logs)和[Traces](https://opentelemetry.io/docs/concepts/observability-primer/#distributed-traces)的持久化采用[Elasticsearch](https://www.elastic.co/cn/elasticsearch/)，[Metrics](https://opentelemetry.io/docs/concepts/observability-primer/#reliability--metrics)持久化采用[Prometheus](https://prometheus.io/)。
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:8]: OpenTelemetry

File: docs/framework/building-blocks/observability.md [hunk 2]
```diff
@@ -1,6 +1,188 @@
...
+
+### Metrics
+ 当前只集成了`Opentelemetry`，采集的`Metric`范围较少，可以根据需求，添加第三方更为成熟的`Metric`监测库，或自定义添加，可参考[Metrics](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics)
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:11]: OpenTelemetry

File: docs/framework/building-blocks/observability.md [hunk 3]
```diff
@@ -1,6 +1,188 @@
...
+
+### Traces
+当前主要集成了http和DataBase(EF core模式)链路追踪，链路相关详细资料可参见[Distributed tracing](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing)和[Collect a distributed trace](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing-collection-walkthroughs?source=recommendations)。
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:14]: EF Core

Comment (doddgu) [inline docs/framework/building-blocks/observability.md:14]: http -> HTTP

File: docs/framework/building-blocks/observability.md [hunk 4]
```diff
@@ -1,6 +1,188 @@
...
+
+#### Logs
+通过集成`Opentelemetry`后，直接使用[ILogger](https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line)记录日志，日志或自动与相关的trace相关联，在通过相关日志排查问题时，可以根据链路traceId查找相关的链路信息，更利于问题排查。
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:17]: OpenTelemetry

Comment (doddgu) [reply docs/framework/building-blocks/observability.md:17]: traceId? TraceId?

Comment (doddgu) [reply docs/framework/building-blocks/observability.md:17]: trace? Trace?

File: docs/framework/building-blocks/observability.md [hunk 5]
```diff
@@ -1,6 +1,188 @@
...
+
+### 功能
+
+1. [更加便捷的集成Opentelemetry，收集可观性数据](#集成Opentelemetry)；
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:21]: 可观测性

Comment (doddgu) [reply docs/framework/building-blocks/observability.md:21]: OpenTelemetry

File: docs/framework/building-blocks/observability.md [hunk 6]
```diff
@@ -1,6 +1,188 @@
...
+```
+
+### 入门
+
+``` C#
+var options = new MasaObservableOptions
+{
+    //服务名称，必须配置
+    ServiceName = "my-project-api",
+    //环境，例如开发、测试、生产等，可不配置
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:40]: 可不配置 -> 可选配置

File: docs/framework/building-blocks/observability.md [hunk 7]
```diff
@@ -1,6 +1,188 @@
...
+    ServiceNameSpace = "development",
+    //服务版本号，可不配置
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:42]: 可不配置 -> 可选配置，默认xxx

File: docs/framework/building-blocks/observability.md [hunk 8]
```diff
@@ -1,6 +1,188 @@
...
+//otlpUrl默认为 "http://localhost:4717"
+builder.Services.AddObservable(builder.Logging, options);
+```
+> [OTLP服务相关部署和配置](https://opentelemetry.io/docs/collector/)
+
+应用程序启动后，该应用在进行可观测性监测时，服务名称为：`my-project-api`，环境为：`development`,服务版本号为：`1.0`,这些参数在服务启动时就已经被确定，运行期间不能被更改，如果想了解更多的参数配置，可参考[Resource Semantic Conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#service)
+
+### 高级
+
+1. 如果想完全自定义可观测性的配置，可以参考[OpenTelemetry for netcore](https://opentelemetry.io/docs/instrumentation/net/getting-started/);
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:54]: netcore -> ASP.NET Core

File: docs/framework/building-blocks/observability.md [hunk 9]
```diff
@@ -1,6 +1,188 @@
...
+
+#### Trace
+
+``` C#
+builder.Services.AddMasaTracing(builder =>
+{
+    var resourceBuilder = ResourceBuilder.CreateDefault();
+    ....
+
+    //balzor应用和api应用在链路的过滤执行条件上有区别
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:91]: blazor

File: docs/framework/building-blocks/observability.md [hunk 10]
```diff
@@ -1,6 +1,188 @@
...
+
+    builder.BuildTraceCallback = options =>
+    {
+        options.SetResourceBuilder(resourceBuilder);
+        options.AddOtlpExporter();
+    };
+});
+```
+
+`resourceBuilder`创建参考[Log](#log)，此处需要明确下，由于在netcore6.0中，[基于`SignalR`的长链接，在链路处理上存在无法区分链路的问题](https://github.com/dotnet/aspnetcore/issues/29846)，当前我们采用了忽略长链接的链路，后续官方版本升级时，我们也会跟进和修正该问题。
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:105]: netcore 6.0 -> .NET 6

File: docs/framework/building-blocks/observability.md [hunk 11]
```diff
@@ -1,6 +1,188 @@
...
+    ....
+
+    builder.SetResourceBuilder(resourceBuilder);    
+    builder.AddRuntimeMetrics();
+    builder.AddOtlpExporter();
+    
+});
+```
+
+该示例集成了`Opentelemetry`并添加了`RuntimeMetrics`运行时`metric`监测，如果需要添加其它`metrics`的监测，自己引入包添加监测即可。
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:122]: OpenTelemetry

File: docs/framework/building-blocks/observability.md [hunk 12]
```diff
@@ -1,6 +1,188 @@
...
+    {
+        options.SetResourceBuilder(resourceBuilder);
+        options.AddOtlpExporter();
+    };
+});
+```
+
+`resourceBuilder`创建参考[Log](#log)，此处需要明确下，由于在netcore6.0中，[基于`SignalR`的长链接，在链路处理上存在无法区分链路的问题](https://github.com/dotnet/aspnetcore/issues/29846)，当前我们采用了忽略长链接的链路，后续官方版本升级时，我们也会跟进和修正该问题。
+
+#### Metric
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:107]: Metric? Metrics?

File: docs/framework/building-blocks/observability.md [hunk 13]
```diff
@@ -1,6 +1,188 @@
...
+当前版本的存储实现为：
+`Logs`和`Traces`存储于`Elasticseach`,`Metrics`存储于`Prometheus`，所以目前版本的Logs和Traces的查询api也是基于Elasticsearch实现。另外使用该功能时，需要引入包：
+
+``` C#
+dotnet add package Masa.Contrib.StackSdks.Tsc.Elasticsearch
+```
+
+### Logs查询
+
+1. `Task<PaginatedListBase<LogResponseDto>> ListAsync(BaseRequestDto query)`：日志列表分页查询，目前最大只能到第9999条数据，
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:136]: Why max 9999?

File: docs/framework/building-blocks/observability.md [hunk 14]
```diff
@@ -1,6 +1,188 @@
...
+示例返回服务名称为"my-project-api"的日志总条数，目前每次查询只支持一级聚合，嵌套聚合目前不支持，聚合类型支持`Count`、`Sum`、`Avg`、`DistinctCount`、`DateHistogram`和`GroupBy`，`Sum`和`Avg`必须为数值类型的字段，`DateHistogram`必须是对日期类型的字段，其它如果是字符串类型，必须为`keyword`类型。
+
+> 返回结果类型：`GroupBy` 返回 `IEnumerable<string>`,`DateHistogram` 返回 `IEnumerable<KeyValuePair<double,string>>`，其它返回`double`。
+
+### Traces查询
+
+1. `Task<IEnumerable<TraceResponseDto>> GetAsync(string traceId)`：根据`traceId`获取整个链路的所有trace信息
+2. `Task<PaginatedListBase<TraceResponseDto>> ListAsync(BaseRequestDto query)`：获取trace列表
+3. `Task<object> AggregateAsync(SimpleAggregateRequestDto query) trace聚合查询`：参见log聚合统计
+4. `void GetAll(BaseRequestDto query, Action<IEnumerable<TraceResponseDto>> resultAction)`：获取指定条件的所有trace数据
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:188]: No real-time and batch settings.

File: docs/framework/concepts/observability.md [hunk 15]
```diff
@@ -3,4 +3,46 @@ title: 可观测性
...
+
+在 IT 和云计算中，可观测性还指用于聚合、关联和分析来自分布式应用程序及其运行的硬件和网络的稳定性能数据流的软件工具和实践，以便更有效地监控、故障排除和调试应用程序和网络以满足客户体验期望、服务水平协议 (SLA) 和其他业务要求。
+
+### 为什么需要可观测性？
+
+在过去时间里，IT 团队主要依靠 APM 和 NPM 来监控应用程序及其运行的网络并对其进行故障排除。
+
+APM系统对应用程序和系统数据进行定期采样和聚合，称为 "遥测"，与已知的应用程序性能问题有关。它分析与关键绩效指标(KPIs)相关的遥测数据，并将结果汇​​总到仪表板中，以提醒运营和支持团队注意应解决的异常情况，以解决或预防问题。
+
+APM 系统对于监视单体应用程序或传统分布式应用程序并对其进行故障排除非常有效，在这些应用程序中，新代码会定期发布，并且应用程序组件、服务器和相关资源之间的工作流和依赖关系是众所周知的或易于跟踪的。
```
Comment (doddgu) [inline docs/framework/concepts/observability.md:22]: APM系统？ APM 系统？

File: docs/framework/concepts/observability.md [hunk 16]
```diff
@@ -3,4 +3,46 @@ title: 可观测性
...
+
+可观测提供了对现代分布式系统的深入可见性，以便更快、自动地识别和解决问题。
+
+一般而言，可观测性是指您仅根据对复杂系统外部输出的了解就可以了解其内部状态或条件的程度。系统的可观测性越强，您就可以越快、越准确地从已识别的性能问题导航到其根本原因，而无需进行额外的测试或编码。
+
+在 IT 和云计算中，可观测性还指用于聚合、关联和分析来自分布式应用程序及其运行的硬件和网络的稳定性能数据流的软件工具和实践，以便更有效地监控、故障排除和调试应用程序和网络以满足客户体验期望、服务水平协议 (SLA) 和其他业务要求。
+
+### 为什么需要可观测性？
+
+在过去时间里，IT 团队主要依靠 APM 和 NPM 来监控应用程序及其运行的网络并对其进行故障排除。
```
Comment (doddgu) [inline docs/framework/concepts/observability.md:18]: What's the concept of APM and NPM?

File: docs/framework/concepts/observability.md [hunk 17]
```diff
@@ -3,4 +3,46 @@ title: 可观测性
...
+
+同样，NPM 系统从混合网络中收集网络性能指标和流量数据，并分析这些数据，以帮助网络运营和工程团队了解网络性能何时不正常。
+
+NPM 系统为组织提供解决网络问题的洞察力，以及规划持续网络性能所需的未来容量所需的数据。
+
+现在企业正在迅速采用现代开发实践——敏捷开发、持续集成 和 持续部署 (CI/CD)、DevOps、多种编程语言——以及 微服务、Docker 容器、Kubernetes 和 无服务器等云原生 技术  功能。因此，他们比以往任何时候都更快地将更多服务推向市场。但在此过程中，他们会如此频繁地在如此多的地方、以如此多的不同语言部署新的应用程序组件，并且部署的时间段差异很大（对于无服务器功能，是几秒钟或几分之一秒），以至于APM 的每分钟一次数据采样无法跟上步伐；同时网络也在不断发展以跟上现代应用程序的需求，对 NPM 系统提出要求，以扩大在日益混合的云环境中收集性能指标和流的方式，并更加以应用程序为中心。
+
+需要的是更高质量的应用程序遥测——以及更多——可用于为每个应用程序用户请求或事务创建高保真、上下文丰富、完全相关的记录，以及更以应用程序为中心的视图应用程序所依赖的网络。
+
+### 可观察性如何运作？
```
Comment (doddgu) [inline docs/framework/concepts/observability.md:32]: 可观测性？可观察性？

File: docs/framework/concepts/observability.md [hunk 18]
```diff
@@ -3,4 +3,46 @@ title: 可观测性
...
+
+应用程序可观察性系统通过与应用程序和基础架构组件中内置的现有仪器相集成，并通过提供工具将仪器添加到这些组件，不断发现和收集应用程序性能遥测数据。应用程序可观察性侧重于四种主要遥测类型、可观察性的三大支柱——`logs`、`metrics`和`traces`——以及依赖性。
```
Comment (doddgu) [inline docs/framework/concepts/observability.md:34]: 可观测性？可观察性？ Twice!

File: docs/framework/concepts/observability.md [hunk 19]
```diff
@@ -3,4 +3,46 @@ title: 可观测性
...
+
+`Logs`是应用程序事件的精细、带时间戳、完整且不可变的记录，采用三种格式：纯文本、结构化或二进制。除其他事项外，日志可用于创建每个事件的高保真度、逐毫秒的记录，包括周围的上下文，开发人员可以“回放”以进行故障排除和调试。
+`Metrics`是给定时间段内应用程序和系统健康状况的基本衡量指标，例如应用程序在五分钟内使用了多少内存或 CPU 容量，或者应用程序在此期间经历了多少延迟使用量激增。
+`Traces`记录了每个用户请求的端到端“旅程”，从 UI 或移动应用程序通过整个分布式架构再回到用户。
+收集这些遥测数据后，平台实时将其关联起来，为DevOps团队、站点可靠性工程(SRE) 团队、软件工程师和其他 IT 人员提供完整的上下文信息——任何事件的内容、地点和原因可能表明，导致或用于解决应用程序性能问题或收集可操作的见解。
+
+许多应用程序可观察性系统会自动发现系统中可能出现的新遥测来源（例如对另一个软件应用程序的新 API 调用）。由于它们处理的数据比标准 APM 解决方案多得多，因此许多平台都包含 AIOps（运营人工智能）功能，可以从噪音（与问题无关的数据）中筛选出信号——实际问题的迹象。
+
+网络可观察性系统采用类似的方法，通过从混合云环境收集一系列网络性能数据。这些数据包括来自路由器和交换机的指标，有助于了解它们的性能和健康状况，以及用于深入了解通过网络设备运行的应用程序的流数据，以及相关的元数据。
```
Comment (doddgu) [inline docs/framework/concepts/observability.md:43]: 可观测性？可观察性？

File: docs/framework/concepts/observability.md [hunk 20]
```diff
@@ -3,4 +3,46 @@ title: 可观测性
...
+### 可观察性如何运作？
+
+应用程序可观察性系统通过与应用程序和基础架构组件中内置的现有仪器相集成，并通过提供工具将仪器添加到这些组件，不断发现和收集应用程序性能遥测数据。应用程序可观察性侧重于四种主要遥测类型、可观察性的三大支柱——`logs`、`metrics`和`traces`——以及依赖性。
+
+`Logs`是应用程序事件的精细、带时间戳、完整且不可变的记录，采用三种格式：纯文本、结构化或二进制。除其他事项外，日志可用于创建每个事件的高保真度、逐毫秒的记录，包括周围的上下文，开发人员可以“回放”以进行故障排除和调试。
+`Metrics`是给定时间段内应用程序和系统健康状况的基本衡量指标，例如应用程序在五分钟内使用了多少内存或 CPU 容量，或者应用程序在此期间经历了多少延迟使用量激增。
+`Traces`记录了每个用户请求的端到端“旅程”，从 UI 或移动应用程序通过整个分布式架构再回到用户。
+收集这些遥测数据后，平台实时将其关联起来，为DevOps团队、站点可靠性工程(SRE) 团队、软件工程师和其他 IT 人员提供完整的上下文信息——任何事件的内容、地点和原因可能表明，导致或用于解决应用程序性能问题或收集可操作的见解。
+
+许多应用程序可观察性系统会自动发现系统中可能出现的新遥测来源（例如对另一个软件应用程序的新 API 调用）。由于它们处理的数据比标准 APM 解决方案多得多，因此许多平台都包含 AIOps（运营人工智能）功能，可以从噪音（与问题无关的数据）中筛选出信号——实际问题的迹象。
```
Comment (doddgu) [inline docs/framework/concepts/observability.md:41]: 可观测性？可观察性？

File: docs/framework/concepts/observability.md [hunk 21]
```diff
@@ -3,4 +3,46 @@ title: 可观测性
...
+
+网络可观察性系统采用类似的方法，通过从混合云环境收集一系列网络性能数据。这些数据包括来自路由器和交换机的指标，有助于了解它们的性能和健康状况，以及用于深入了解通过网络设备运行的应用程序的流数据，以及相关的元数据。
+
+### Opentelemtry
+[`OpenTelemetry`](https://opentelemetry.io/)，也简称为`OTel`，是一个供应商中立、开源的可观察性框架，用于检测、生成、收集和导出遥测数据，如 `traces`、`metrics`和`logs`。作为一项行业标准，它受到许多供应商的原生支持。
```
Comment (doddgu) [inline docs/framework/concepts/observability.md:46]: 可观测性？可观察性？

File: docs/framework/concepts/observability.md [hunk 22]
```diff
@@ -3,4 +3,46 @@ title: 可观测性
...
+`Logs`是应用程序事件的精细、带时间戳、完整且不可变的记录，采用三种格式：纯文本、结构化或二进制。除其他事项外，日志可用于创建每个事件的高保真度、逐毫秒的记录，包括周围的上下文，开发人员可以“回放”以进行故障排除和调试。
+`Metrics`是给定时间段内应用程序和系统健康状况的基本衡量指标，例如应用程序在五分钟内使用了多少内存或 CPU 容量，或者应用程序在此期间经历了多少延迟使用量激增。
+`Traces`记录了每个用户请求的端到端“旅程”，从 UI 或移动应用程序通过整个分布式架构再回到用户。
+收集这些遥测数据后，平台实时将其关联起来，为DevOps团队、站点可靠性工程(SRE) 团队、软件工程师和其他 IT 人员提供完整的上下文信息——任何事件的内容、地点和原因可能表明，导致或用于解决应用程序性能问题或收集可操作的见解。
+
+许多应用程序可观察性系统会自动发现系统中可能出现的新遥测来源（例如对另一个软件应用程序的新 API 调用）。由于它们处理的数据比标准 APM 解决方案多得多，因此许多平台都包含 AIOps（运营人工智能）功能，可以从噪音（与问题无关的数据）中筛选出信号——实际问题的迹象。
+
+网络可观察性系统采用类似的方法，通过从混合云环境收集一系列网络性能数据。这些数据包括来自路由器和交换机的指标，有助于了解它们的性能和健康状况，以及用于深入了解通过网络设备运行的应用程序的流数据，以及相关的元数据。
+
+### Opentelemtry
```
Comment (doddgu) [inline docs/framework/concepts/observability.md:45]: OpenTelemetry

File: docs/framework/utils/data/prometheus.md [hunk 23]
```diff
@@ -0,0 +1,129 @@
...
+1. 安装包
+
+``` C#
+dotnet add package Masa.Utils.Data.Prometheus
+```
+
+2. 服务注册
+
+``` C#
+var prometheusUrl="http://localhost:9090";
```
Comment (doddgu) [inline docs/framework/utils/data/prometheus.md:31]: Format code.

File: docs/framework/utils/data/prometheus.md [hunk 24]
```diff
@@ -0,0 +1,129 @@
...
+``` C#
+public class MyPrometheusQueryService
+{
+    private readonly IMasaPrometheusClient _client;
+
+    public MyPrometheusQueryService(IMasaPrometheusClient client)
+    {
+        _client = client;
+    }
+
```
Comment (doddgu) [inline docs/framework/utils/data/prometheus.md:46]: Remove the blank line.

**Review by doddgu — CHANGES_REQUESTED** [review]

File: docs/framework/building-blocks/observability.md [hunk 25]
```diff
@@ -5,20 +5,20 @@ date: 2022/12/07
...
 
 ### Metrics
- 当前只集成了`Opentelemetry`，采集的`Metric`范围较少，可以根据需求，添加第三方更为成熟的`Metric`监测库，或自定义添加，可参考[Metrics](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics)
+ 当前只集成了`OpenTelemetry`，采集的`Metrics`范围较少，可以根据需求，添加第三方更为成熟的`Metrics`监测库，或自定义添加，可参考[Metrics](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics)
 
 ### Traces
-当前主要集成了http和DataBase(EF core模式)链路追踪，链路相关详细资料可参见[Distributed tracing](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing)和[Collect a distributed trace](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing-collection-walkthroughs?source=recommendations)。
+当前主要集成了`Http`和`Database`(EF core模式)链路追踪，链路相关详细资料可参见[Distributed tracing](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing)和[Collect a distributed trace](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing-collection-walkthroughs?source=recommendations)。
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md]: EF core -> EF Core

File: docs/framework/building-blocks/observability.md [hunk 26]
```diff
@@ -45,16 +45,16 @@ var options = new MasaObservableOptions
...
-> [OTLP服务相关部署和配置](https://opentelemetry.io/docs/collector/)
+> [OTLP服务相关部署和配置](https://OpenTelemetry.io/docs/collector/)
 
-应用程序启动后，该应用在进行可观测性监测时，服务名称为：`my-project-api`，环境为：`development`,服务版本号为：`1.0`,这些参数在服务启动时就已经被确定，运行期间不能被更改，如果想了解更多的参数配置，可参考[Resource Semantic Conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#service)
+应用程序启动后，该应用在进行可观测性监测时，服务名称为：`my-project-api`，环境为：`development`,服务版本号为：`1.0`,这些参数在服务启动时就已经被确定，运行期间不能被更改，如果想了解更多的参数配置，可参考[Resource Semantic Conventions](https://github.com/open-telemetry/OpenTelemetry-specification/tree/main/specification/resource/semantic_conventions#service)
 
 ### 高级
 
-1. 如果想完全自定义可观测性的配置，可以参考[OpenTelemetry for netcore](https://opentelemetry.io/docs/instrumentation/net/getting-started/);
+1. 如果想完全自定义可观测性的配置，可以参考[OpenTelemetry for netcore](https://OpenTelemetry.io/docs/instrumentation/net/getting-started/);
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:54]: OpenTelemetry -> opentelemetry

File: docs/framework/building-blocks/observability.md [hunk 27]
```diff
@@ -45,16 +45,16 @@ var options = new MasaObservableOptions
 //otlpUrl默认为 "http://localhost:4717"
 builder.Services.AddObservable(builder.Logging, options);
 ```
-> [OTLP服务相关部署和配置](https://opentelemetry.io/docs/collector/)
+> [OTLP服务相关部署和配置](https://OpenTelemetry.io/docs/collector/)
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:48]: OpenTelemetry -> opentelemetry

File: docs/framework/building-blocks/observability.md [hunk 28]
```diff
@@ -5,20 +5,20 @@ date: 2022/12/07
 
 ## 概念
 
-我们基于[Opentelemetry](https://opentelemetry.io/docs/)的可观测性接入，使用标准的OTLPExporter，[Logs](https://opentelemetry.io/docs/concepts/observability-primer/#logs)和[Traces](https://opentelemetry.io/docs/concepts/observability-primer/#distributed-traces)的持久化采用[Elasticsearch](https://www.elastic.co/cn/elasticsearch/)，[Metrics](https://opentelemetry.io/docs/concepts/observability-primer/#reliability--metrics)持久化采用[Prometheus](https://prometheus.io/)。
+我们可观测性基于[OpenTelemetry](https://OpenTelemetry.io/docs/)的接入，使用标准的OTLPExporter，[Logs](https://OpenTelemetry.io/docs/concepts/observability-primer/#logs)和[Traces](https://OpenTelemetry.io/docs/concepts/observability-primer/#distributed-traces)的持久化采用[Elasticsearch](https://www.elastic.co/cn/elasticsearch/)，[Metrics](https://OpenTelemetry.io/docs/concepts/observability-primer/#reliability--metrics)持久化采用[Prometheus](https://prometheus.io/)。
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:8]: url：OpenTelemetry -> opentelemetry

File: docs/framework/building-blocks/observability.md [hunk 29]
```diff
@@ -185,4 +185,8 @@ var result = await _logService.AggregateAsync(query);
 1. `Task<IEnumerable<TraceResponseDto>> GetAsync(string traceId)`：根据`traceId`获取整个链路的所有trace信息
 2. `Task<PaginatedListBase<TraceResponseDto>> ListAsync(BaseRequestDto query)`：获取trace列表
 3. `Task<object> AggregateAsync(SimpleAggregateRequestDto query) trace聚合查询`：参见log聚合统计
-4. `void GetAll(BaseRequestDto query, Action<IEnumerable<TraceResponseDto>> resultAction)`：获取指定条件的所有trace数据
\ No newline at end of file
+4. `void GetAll(BaseRequestDto query, Action<IEnumerable<TraceResponseDto>> resultAction)`：获取指定条件的所有trace数据
+
+### Metrics查询
+
+目前采用的标准[`Http api`](../utils/data/prometheus.md)，目前仅实现了几个常用的api。
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md]: api -> API

File: docs/framework/building-blocks/observability.md [hunk 30]
```diff
@@ -119,13 +119,13 @@ builder.Services.AddMasaMetrics(builder =>
...
 
-该示例集成了`Opentelemetry`并添加了`RuntimeMetrics`运行时`metric`监测，如果需要添加其它`metrics`的监测，自己引入包添加监测即可。
+该示例集成了`OpenTelemetry`并添加了`RuntimeMetrics`运行时`metric`监测，如果需要添加其它`metrics`的监测，自己引入包添加监测即可。
 
 
 ## 遥测数据查询
 
 当前版本的存储实现为：
-`Logs`和`Traces`存储于`Elasticseach`,`Metrics`存储于`Prometheus`，所以目前版本的Logs和Traces的查询api也是基于Elasticsearch实现。另外使用该功能时，需要引入包：
+`Logs`和`Traces`存储于`Elasticseach`,`Metrics`存储于`Prometheus`，所以目前版本的`Logs`和`Traces`的查询api也是基于`Elasticsearch`实现。另外使用该功能时，需要引入包：
```
Comment (doddgu) [inline docs/framework/building-blocks/observability.md:128]: api -> API

**Review by doddgu — CHANGES_REQUESTED** [review]

File: docs/framework/building-blocks/observability.md (hunk 14 above)
Comment (Qinyouzeng) [reply docs/framework/building-blocks/observability.md]: `Traces` and `Logs` not has real-time and batch settings API, `Metrics` has these API.

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**